# Conversion Rate Challenge
## Data Science Directed Study
### michelle.carney@berkeley.edu

We have data about users who hit our site: whether they converted or not as well as some of their characteristics such as their country, the marketing channel, their age, whether they are repeated users and the number of pages visited during that session (as a proxy for site activity/time spent on the site).

Your project is to:
 - Predict Conversion Rate (# conversions / total sessions)
 - Come up with recommendations for the product team and the marketing team to improve the conversion rate
 
 
Columns:

- country : user country based on the IP address

- age : user age. Self-reported at sign-in step

- new_user : whether the user created the account during this session or had already an account and simply came back to the site

- source : marketing channel source

  - Ads: came to the site by clicking on an advertisement
  - Seo: came to the site by clicking on search results
  - Direct: came to the site by directly typing the URL on the browser

- total_pages_visited: number of total pages visited during the session. This is a proxy for time spent on site and engagement during the session.

- converted: this is our label. 1 means they converted within the session, 0 means they left without buying anything. The company goal is to increase conversion rate: # conversions / total sessions.

In [1]:
# Let's start with our imports

import pandas as pd
import numpy as np
import random

In [13]:
# Let's load the data and investigate in pandas

df = pd.read_csv("conversion_data.csv")
df.tail()
# len(df)

country  age  new_user  source  total_pages_visited  converted
316195      US   26         1     Ads                    9          0
316196      US   27         1  Direct                    6          0
316197   China   25         0     Ads                    4          0
316198      US   22         0     Seo                    4          0
316199      UK   24         0     Ads                    4          0

We can see that there are 316200 rows, or 316199 individual entries

From the 